crates/ruff_python_literal/src/format.rs

```diff
@@ -206,6 +206,8 @@ pub struct FormatSpec {
     precision: Option<usize>,
     // Ex) `f` in `'{:+f}'`
     format_type: Option<FormatType>,
+    // Ex) `f` in `'{1:{f}}'`
```

Comment:
My understanding is that different components of the spec can use nested parts. For example, this seems to be valid:

```python
>>> "{0}{1:*^{width}}".format("a", "1", width=15)
'a*******1*******'
```

Is the approach here limited to nested `format_type`?

Reply:
I don't know which of the components are allowed to be nested. It might be _any_ of them? E.g., this works too:

```python
>>> "{0}{1:*{width}15}".format("a", "1", width="^")
'a*******1*******'
```

Reply:
Ah, so does this:

```python
>>> "{0}{1:{width}15}".format("a", "1", width="*^")
'a*******1*******'
```

So it may not even be possible to capture this with this data model, since we can't map from part to nested expression.